src/textprop.c

```diff
@@ -536,6 +554,193 @@ f_prop_clear(typval_T *argvars, typval_T *rettv UNUSED)
     redraw_buf_later(buf, NOT_VALID);
 }
 
+/*
+ * prop_find({props} [, {direction}])
+ */
+    void
+f_prop_find(typval_T *argvars, typval_T *rettv)
+{
+    int		lnum_start;
+    int		start_pos_has_prop = 0;
+    int		seen_end = 0;
+    int		id = -1;
+    int		type_id = -1;
+    int		skipstart = 0;
+    int		lnum = -1;
+    int		col = -1;
+    pos_T	*cursor = &curwin->w_cursor;
+    char_u	*dir_s;
+    int		dir = 1;    // 1 = forward, -1 = backward
+    dict_T	*dict;
+    dictitem_T	*di;
+    buf_T	*buf = curbuf;
+
+    if (argvars[0].v_type != VAR_DICT || argvars[0].vval.v_dict == NULL)
+    {
+	emsg(_(e_invarg));
+	return;
+    }
+
+    dict = argvars[0].vval.v_dict;
+    
+    if (get_bufnr_from_arg(&argvars[0], &buf) == FAIL)
+	return; 
+    if (buf->b_ml.ml_mfp == NULL)
+	return;
+
+    if (argvars[1].v_type != VAR_UNKNOWN)
+    {
+	dir_s = tv_get_string(&argvars[1]);
+	if (*dir_s == 'b') 
+	    dir = -1;
+	else if (*dir_s != 'f')
+	{
+	    emsg(_(e_invarg));
+	    return;
+	}
+    }
+
+    di = dict_find(dict, (char_u *)"lnum", -1);
+    if (di != NULL)
+	lnum = tv_get_number(&di->di_tv);
+
+    di = dict_find(dict, (char_u *)"col", -1);
+    if (di != NULL)
+	col = tv_get_number(&di->di_tv);
+
+    if (lnum == -1)
+    {
+	lnum = cursor->lnum;
+	col = cursor->col + 1;
+    }
+    else    
+    {
+	if (col == -1)
+	    col = 1;
+    }
+
+    if (lnum < 1 || lnum > buf->b_ml.ml_line_count)
+    {
+	emsg(_(e_invrange));
+	return;
+    }
+    
+    di = dict_find(dict, (char_u *)"skipstart", -1);
+    if (di != NULL)
+	skipstart = tv_get_number(&di->di_tv);
+
+    if (dict_find(dict, (char_u *)"id", -1) != NULL)
+	id = dict_get_number(dict, (char_u *)"id");
+    if (dict_find(dict, (char_u *)"type", -1))
+    {
+	char_u	    *name = dict_get_string(dict, (char_u *)"type", FALSE);
+	proptype_T  *type = lookup_prop_type(name, buf);
+
+	if (type == NULL)
+	    return;
+	type_id = type->pt_id;
+    }
+    if (id == -1 && type_id == -1)
+    {
+	emsg(_("E968: Need at least one of 'id' or 'type'"));
+	return;
+    }
+
+    lnum_start = lnum;
+
+    if (rettv_dict_alloc(rettv) == FAIL)
+	return;
+    
+    while (1)
+    {	
+        char_u	*text = ml_get_buf(buf, lnum, FALSE);
```

Comment:
Indent of this `while` block looks wrong.

Reply:
Recreated the indents for prop_find, they should all be consistent. Not sure what happened here but looks good now.